EthWorks/UniversalLoginSDK
# [#76] Full account recovery with backup codes.

This is the full account recovery flow hooked together. 

To test:
1. Sign into an account
2. Generate backup codes
3. Copy one of the backup codes
4. Attempt log into alternative device
5. Enter one-time code you copied.

The UI should be changed for this (currently the button labels change and nothing else @alexvandesande  perhaps you can help with this.)

I'm going to leave the other pull request up so that if we decide on a different flow for hooking everything together we can work with that branch.

**Question**
Are backup codes one time use? If yes I suggest:
1. Adding a keySwap method to the keyHolder.sol which adds and removes a key from the contract in one transaction (Adds device key, removes recovery key)
2. Making it clearer to the user that these are one time codes.
